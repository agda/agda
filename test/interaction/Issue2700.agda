open import Agda.Builtin.Nat

record Pointed (A : Set) : Set where
 field point : A

it : ∀ {A : Set} {{x : A}} → A
it {{x}} = x

instance _ = record { point = 3 - 4 }

_ : Pointed Nat
_ = {!!}

-- Andreas, 2025-04-01, issue #7624, testcase by Szumi Xie
-- cmd_goal_type_context_check should also work if the expression introduces new metas

_ : Set
_ = {! ? !}
