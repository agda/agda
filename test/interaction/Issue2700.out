(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Pointed Nat ?1 : Set " nil)
((last . 1) . (agda2-goals-action '(0 1)))
(agda2-status-action "")
(agda2-info-action "*Goal type etc.*" "Goal: Pointed Nat Elaborates to: record { point = 0 } ————————————————————————————————————————————————————————————" nil)
(agda2-status-action "")
(agda2-info-action "*Goal type etc.*" "Goal: Set Elaborates to: ?2 ————————————————————————————————————————————————————————————" nil)
